src/selectorNode/SelectorNode.js

```diff
@@ -0,0 +1,86 @@
+import { svgns } from '../canvas/Canvas';
+
+export class SelectorNode {
+    constructor(canvas, location, id, eventListeners, svgTag="circle", attributes={}) {
+
+        /* TODOS: ADD EVENT LISTENERS */
+
+        this.domElement = document.createElementNS(svgns, svgTag);
+
+        // styling the node
+        // STYLING THE NODE IS DONE ONLY ONCE; HENCE NO FUNCTIONALITY TO CHANGE IT
+        for (let attribute in attributes) {
+            this.domElement.setAttributeNS(null, attribute, attributes[attribute]);
+        }
+
+        this.canvas = canvas; // canvas of which the node is a part of
+        this.location = location; // where the node is located
+        
+        // setting the location of the node
+        this.domElement.setAttributeNS(null, "cx", this.location.x);
+        this.domElement.setAttributeNS(null, "cy", this.location.y);
+
+        this.canvas.nodesContainer.appendChild(this.domElement); // adding the node to the nodes container
+
+        this.shapes = []; // list of shapes that the node is associated with
+        this.connections = []; // list of other node connections
+
+        this.eventListeners = [];
+
+        // DEEP COPY OF THE EVENT LISTENER OBJECTS
+        // if this is not done, multiple binding of functions takes place
+        // result = sad 1ntEgr8 :(
+        for (let i = 0; i < eventListeners.length; i++) {
+            this.eventListeners.push({
```

Comment:
I'm sure there's a Object.copy() you could use here

Reply:
Use lodash

Reply:
i tried Object.copy and it didn't work, but maybe I was using it the wrong way. 

I'll look into lodash